src/runtime/buffer_t.cpp

```diff
@@ -153,10 +154,60 @@ halide_buffer_t *_halide_buffer_crop(halide_buffer_t *dst,
         dst->dim[i].extent = extent[i];
         offset += (min[i] - src->dim[i].min) * src->dim[i].stride;
     }
-    dst->host += offset * src->type.bytes();
+    offset *= src->type.bytes();
+    dst->host += offset;
```

Comment:
We need to conditionalize this on non-null host allocation. (Can show up for device only buffers, but possibly also for lazy allocated ones.) If not, it should probably assert non-null. Adding to null pointers makes for difficult to track down bugs. (Also: https://github.com/halide/Halide/issues/2058 .)